src/main/java/org/opensearch/flowframework/common/FlowFrameworkSettings.java

```diff
@@ -82,27 +85,31 @@ public FlowFrameworkSettings(ClusterService clusterService, Settings settings) {
         // Currently this is just an on/off switch for the entire plugin's API.
         // If desired more fine-tuned feature settings can be added below.
         this.isFlowFrameworkEnabled = FLOW_FRAMEWORK_ENABLED.get(settings);
-        this.maxRetry = MAX_GET_TASK_REQUEST_RETRY.get(settings);
+        this.retryDuration = TASK_REQUEST_RETRY_DURATION.get(settings);
         this.maxWorkflowSteps = MAX_WORKFLOW_STEPS.get(settings);
+        this.maxWorkflows = MAX_WORKFLOWS.get(settings);
+        this.requestTimeout = WORKFLOW_REQUEST_TIMEOUT.get(settings);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(FLOW_FRAMEWORK_ENABLED, it -> isFlowFrameworkEnabled = it);
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(MAX_GET_TASK_REQUEST_RETRY, it -> maxRetry = it);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(TASK_REQUEST_RETRY_DURATION, it -> retryDuration = it);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(MAX_WORKFLOW_STEPS, it -> maxWorkflowSteps = it);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(MAX_WORKFLOWS, it -> maxWorkflows = it);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(WORKFLOW_REQUEST_TIMEOUT, it -> requestTimeout = it);
```

Comment:
Shouldn't we have these settings in the AbstractClass which was deleted below? Reason being the listener for both the settings would be listening only when a REST call is invoked rather during plugin initialization.  Trying to understand more here.

Reply:
> Shouldn't we have these settings in the AbstractClass which was deleted below?

"Should"? Probably no.  "Could?" Maybe, but that adds unnecessary transport layer comms.

Putting them in the Abstract class means you'll end up registering two different update consumers, one for each subclass instance.  Not that big a deal in this case as we know we're only instantiating each RestAction once, and updates are probably rare, but still redundant. 

Putting them in the RestAction means you have to communicate them to the TransportAction in the Request, so we had extra parameters in the WorkflowRequest.

Having them all in one place, in an injected class, makes them easy to access directly in the TransportAction and simplifies the WorkflowRequest.

Reply:
Makes sense. That helps.